libs/core/algorithms/include/hpx/parallel/algorithms/detail/adjacent_difference.hpp

```diff
@@ -27,13 +28,24 @@ namespace hpx { namespace parallel { inline namespace v1 { namespace detail {
             sequential_adjacent_difference_t<ExPolicy>>
     {
     private:
-        template <typename InIter, typename OutIter, typename Op>
+        template <typename InIter, typename Sent, typename OutIter, typename Op>
         friend inline OutIter tag_fallback_invoke(
-            sequential_adjacent_difference_t<ExPolicy>, InIter first,
-            InIter last, OutIter dest, Op&& op)
+            sequential_adjacent_difference_t<ExPolicy>, InIter first, Sent last,
+            OutIter dest, Op&& op)
         {
-            return std::adjacent_difference(
-                first, last, dest, std::forward<Op>(op));
+            if (first == last)
+                return dest;
+
+            using value_t = typename std::iterator_traits<InIter>::value_type;
+            value_t acc = *first;
+            *dest = acc;
+            while (++first != last)
+            {
+                value_t val = *first;
+                *++dest = op(val, std::move(acc));    // std::move since C++20
```

Comment:
```suggestion
                *++dest = HPX_INVOKE(op, val, std::move(acc));
```
What is the `C++20` comment saying?

Reply:
I removed this C++20 comment